src/client.ts

```diff
@@ -72,7 +62,9 @@ export class SessionBackend {
           console.error(`Error parsing status stream message as JSON: "${text}"`, e)
         }
         if (data?.state === 'Ready') {
-          this.openSocket()
+          this._isReady = true
+          this._onReady.forEach((cb) => cb())
+          this._onReady = []
```

Comment:
Nit: since we're duplicating this logic, maybe we should factor it out into its own private method like `setIsReady` or `markIsReady` or something like that?